packages/error/src/index.ts

```diff
@@ -53,3 +53,29 @@ export function newMessage(err: any, newMsg: string): ErrorWithData {
   newErr.message = `${newMsg}: ${newErr.message}`;
   return newErr;
 }
+
+/**
+ * Create a new error based an existing one, optionally adding status, message,
+ * and/or data. The existing error will be set as the `cause` of the new error.
+ *
+ * @param err An existing error.
+ * @param param.status Status code to set on the new error.
+ * @param param.message Message to add to the new error.
+ * @param param.data Data to set on the new error.
+ * @returns
```

Comment:
```suggestion
 * @returns The augmented error.
```

Reply:
Done in https://github.com/PrairieLearn/PrairieLearn/pull/7968/commits/668038b531bc71515f37dafd5facf2e80287e1e2